app/controllers/assets_controller.rb

```diff
@@ -23,10 +23,18 @@ def latest
     end
   end
 
+  def set_related_user_variables
+    all_user_tracks = Asset.for_user(@user.id)
```

Comment:
We can get rid of this line and below instead of using `all_user_tracks` just do `@user.assets.hottest` etc...